src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+        private DeltaDeletedEntityObject<TStructuralType> CreateDeletedEntityForFailedOperation(Delta<TStructuralType> changedObj)
+        {
+            Type type = typeof(DeltaDeletedEntityObject<>).MakeGenericType(changedObj.ExpectedClrType);
+
+            DeltaDeletedEntityObject<TStructuralType> deletedObject = Activator.CreateInstance(type, true, changedObj.InstanceAnnotationsPropertyInfo) as DeltaDeletedEntityObject<TStructuralType>;
+
+            foreach (string property in changedObj.GetChangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            foreach (string property in changedObj.GetUnchangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            object annValue;
+            changedObj.TryGetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, out annValue);
+
+            IODataInstanceAnnotationContainer instanceAnnotations = annValue as IODataInstanceAnnotationContainer;
+
+            if (instanceAnnotations != null)
+            {
+                deletedObject.TrySetPropertyValue(changedObj.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+            }
+
+            deletedObject.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
+
+            TryGetContentId(changedObj, _keys, deletedObject);
+            
+            return deletedObject;
+        }
+
+        private static void TryGetContentId(Delta<TStructuralType> changedObj, IList<string> keys, DeltaDeletedEntityObject<TStructuralType> edmDeletedObject)
+        {
+            bool takeContentId = false;
+            for (int i = 0; i < keys.Count; i++)
+            {
+                object value;
+                edmDeletedObject.TryGetPropertyValue(keys[i], out value);
+
+                if (value == null)
+                {
+                    takeContentId = true;
+                    break;
+                }
+            }
+
+            if (takeContentId)
+            {
+                object contentId = changedObj.TransientInstanceAnnotationContainer.GetResourceAnnotation("Core.ContentID");
+                if (contentId != null)
+                {
+                    edmDeletedObject.Id = new Uri(contentId.ToString());
+                }
+                else
+                {
+                    edmDeletedObject.Id = new Uri(string.Empty);
+                }
+            }
+        }
+
+        private static void SetPropertyValues(Delta<TStructuralType> changedObj, DeltaDeletedEntityObject<TStructuralType> edmDeletedObject, string property)
+        {
+            object objectVal;
+            if (changedObj.TryGetPropertyValue(property, out objectVal))
+            {
+                edmDeletedObject.TrySetPropertyValue(property, objectVal);
+            }
+        }
+
+        internal ICollection<TStructuralType> GetInstance()
```

Comment:
>GetInstance [](start = 46, length = 11)

Is this used anywhere? Note that this returns a new instance of a collection, which is different than GetInstance() on Delta<T> which returns the underlying CLR instance.

Unless we need this, we should remove it.